packages/workflow-core/src/lib/plugins/external-plugin/bank-account-verification-plugin/bank-account-verification-plugin.ts

```diff
@@ -0,0 +1,221 @@
+    city: z.string().min(1),
+    postcode: z.string().min(1),
+  }),
+  bankAccountDetails: z.object({
+    holder: z.union([
+      z.union([
+        z.object({
+          bankAccountName: z.string().min(1),
+          companyRegistrationNumber: z.string().min(1),
+        }),
+        z.object({
+          bankAccountName: z.string().min(1),
+          registeredCharityNumber: z.string().min(1),
+        }),
+      ]),
+      z.object({
+        firstName: z.string().min(1),
+        middleName: z.string().optional(),
+        lastName: z.string().min(1),
+      }),
+    ]),
+    sortCode: z.string().min(1),
+    bankAccountNumber: z.string().min(1),
+  }),
+});
+
+type TBankAccountVerificationPluginPayload = {
+  clientId: PluginPayloadProperty;
+  vendor: PluginPayloadProperty;
+  data: {
+    address: {
+      streetNumber: PluginPayloadProperty;
+      street: PluginPayloadProperty;
+      city: PluginPayloadProperty;
+      postcode: PluginPayloadProperty;
+    };
+    bankAccountDetails: {
+      sortCode: PluginPayloadProperty;
+      bankAccountNumber: PluginPayloadProperty;
+    } & (
+      | {
+          holder: {
+            firstName: PluginPayloadProperty;
+            middleName: PluginPayloadProperty<string | undefined>;
+            lastName: PluginPayloadProperty;
+          };
+        }
+      | ({ bankAccountName: PluginPayloadProperty } & (
+          | { companyRegistrationNumber: PluginPayloadProperty }
+          | { registeredCharityNumber: PluginPayloadProperty }
+        ))
+    );
+  };
+};
+
+export class BankAccountVerificationPlugin extends ApiPlugin {
+  public static pluginType = 'http';
+  public payload: TBankAccountVerificationPluginPayload;
+
+  private pluginName = 'Bank Account Verification Plugin';
+
+  constructor({
+    payload,
+    ...pluginParams
+  }: IApiPluginParams & { payload: BankAccountVerificationPlugin['payload'] }) {
+    super({
+      ...pluginParams,
+      response: {
+        ...pluginParams.response,
+        transformers: [
+          ...(pluginParams.response?.transformers ?? []),
+          getTransformer({
+            mapping: [
+              {
+                method: 'setTimeToRecordUTC',
+                source: 'invokedAt',
+                target: 'invokedAt',
+              },
+            ],
+            transformer: 'helper',
+          }),
+        ],
+      },
+      method: 'POST' as const,
+    });
+
+    this.payload = payload;
+
+    handleJmespathTransformers({
+      pluginName: this.pluginName,
+      requestTransformers: this.request?.transformers,
+      responseTransformers: this.response?.transformers,
+    });
+  }
+
+  async invoke(context: TContext) {
+    const env = validateEnv(this.pluginName);
+
+    let requestPayload;
+
+    if (this.request?.transformers) {
+      requestPayload = await this.transformData(this.request.transformers, context);
+
+      const { isValidRequest, errorMessage } = await this.validateContent(
```

Comment:
Why would a static plugin like that will have request/reponse schema coming from configuration?